src/Uno.Extensions.Navigation.UI/FrameworkElementExtensions.cs

```diff
@@ -55,12 +54,18 @@ public static Task HostAsync(
 			{
 				start = () => nav.NavigateViewModelAsync(root, initialViewModel);
 			}
+			else
+			{
+				start = () => nav.NavigateRouteAsync(root, initialRoute ?? string.Empty);
+			}
+
 			var fullstart = async () =>
 			{
-				await initialNavigation();
 				await start();
```

Comment:
Seems like it makes sense but I feel like maybe we are relying on an initial nav to "" to set certain things up maybe?

Will look at bit deeper in the navigation init code but maybe something @nickrandolph can tell us off the top of his head?

Reply:
Not sure if this is the right way of fixing this actually. If someone tries to navigate to a specific url like "localhost/Main" directly, it would not initialize the `ShellViewModel`.